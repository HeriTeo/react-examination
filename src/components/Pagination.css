:local(.paginationBtn) {
  	display: flex;
  	color: white;
  	padding: 10px;
}

:local(.paginationBtn button) {
	margin: 5px 0px 5px 0px;
	background-color: #fff;
	color: #5c75b0;
	border: 1px solid #5c75b0;
}