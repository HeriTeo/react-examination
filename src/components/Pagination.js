import React, { Component } from 'react';
import styles from './Pagination.css';

class Pagination extends Component {
  render() {
    const { paginationPage, clickPage } = this.props;
    return (
      <div className={styles.paginationBtn}>
        {paginationPage.map((players, idx) => (
          <button onClick={() => clickPage(idx)} key={idx}> {idx+1} </button>
        ))}
      </div>
    );
  }
}

Pagination.defaultProps = {
  paginationPage: [[{}]],
};

export default Pagination;
