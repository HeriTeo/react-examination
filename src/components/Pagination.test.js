import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import Pagination from './Pagination';

configure({ adapter: new Adapter() });

describe('Pagination', () => {
  test('renders without crashing', () => {
    const wrapper = shallow(<Pagination />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
