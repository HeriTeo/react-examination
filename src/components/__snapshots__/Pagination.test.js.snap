// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders without crashing 1`] = `
<div>
  <button
    key="0"
    onClick={[Function]}
  >
     
    1
     
  </button>
</div>
`;
